x/slashing/keeper/infractions.go

```diff
@@ -80,6 +85,7 @@ func (k Keeper) HandleValidatorSignatureConcurrent(ctx sdk.Context, addr cryptot
 	maxMissed := k.SignedBlocksWindow(ctx) - minSignedPerWindow
 	shouldSlash = false
 	// if we are past the minimum height and the validator has missed too many blocks, punish them
+	// TODO: this may be problematic if the signed blocks window was just reduced -> then missed blocks counter might be too high due to historical misses
```

Comment:
If the window size changed, then when we do evictions, we will not count the old historical blocks, so this should not be a concern right?

Reply:
correct, removed the TODO,  i had initially put it there since i wasnt certain based on the old prior approach